ios/RNCardConnectReactLibrary.m

```diff
@@ -37,15 +103,216 @@ - (dispatch_queue_t)methodQueue
     }];
 }
 
+RCT_EXPORT_METHOD(discoverDevice) {
+
+    if (_swiper.connectionState == BMSSwiperConnectionStateSearching) {
+        [_swiper cancelFindDevices];
+    }
+    else if (_swiper.connectionState == BMSSwiperConnectionStateConnecting) {
+        [_swiper releaseDevice];
+    }
+
+    [self.swiper findDevices];
+}
+
+RCT_EXPORT_METHOD(activateDevice) {
+
+    if (self.restartReaderBlock) {
+        self.restartReaderBlock();
+        self.restartReaderBlock = nil;
+    }
+}
+
+RCT_EXPORT_METHOD(cancelTransaction) {
+
+    [self.swiper cancelTransaction];
+}
+
+RCT_EXPORT_METHOD(connectToDevice:(NSString *)uuid) {
+
+    if (_swiper.connectionState == BMSSwiperConnectionStateSearching) {
+        [_swiper cancelFindDevices];
+    }
+
+    NSUUID *converted = [[NSUUID alloc] initWithUUIDString:uuid];
+
+    [_swiper connectToDevice:converted mode:BMSCardReadModeSwipeDipTap];
+
+    self.isConnecting = true;
+}
+
+- (void) debug:(NSString *)message {
+
+    if (self.enableDebugging) {
+        [self sendEventWithName:@"BoltOnLogUpdate" body:@{@"debugMessage": message}];
+    }
+}
+
+- (void)swiper:(BMSSwiperController *)swiper configurationProgress:(float)progress {
+
+    [self sendEventWithName:@"BoltOnDeviceConfigurationProgressUpdate" body:@{@"progress": [NSNumber numberWithFloat:progress]}];
+}
+
+- (void)swiper:(BMSSwiperController *)swiper displayMessage:(NSString *)message canCancel:(BOOL)cancelable {
+
+    // special message we will convert into various events
+    if ([message isEqualToString:@"PLEASE SWIPE,\nTAP, OR INSERT"]) {
```

Comment:
Interesting!  I wonder if codifying this special value into a constant might feel good on the ole brain.